apps/aevm/src/aevm_ae_primops.erl

```diff
@@ -260,3 +325,28 @@ get_args(Types, Data) ->
     [_ | Args] = tuple_to_list(Val),
     Args.
 
+no_dynamic_cost(Cb) when is_function(Cb, 0) ->
+    {ok, _DynCost=0, Cb}.
+
+chain_ttl_delta(TTL, State) ->
+    ChainHeight = call_chain1(fun(API, ChainState) -> API:get_height(ChainState) end, State),
+    ttl_delta(ChainHeight, TTL).
+
+ttl_delta(_, {delta, D}) when
```

Comment:
This looks more or less identical to `aeo_util:ttl_delta/2` what motivates having a duplicate here?

Reply:
No exception (`error(...)`).
Stricter control on domain and hence guarantees on return value.